src/main/java/org/elasticsearch/search/aggregations/reducers/ContiguousBucketHelpers.java

```diff
@@ -0,0 +1,109 @@
+package org.elasticsearch.search.aggregations.reducers;
+
+import org.elasticsearch.ElasticsearchIllegalStateException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.SearchParseException;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+import org.elasticsearch.search.aggregations.InvalidAggregationPathException;
+import org.elasticsearch.search.aggregations.bucket.histogram.InternalHistogram;
+import org.elasticsearch.search.aggregations.metrics.InternalNumericMetricsAggregation;
+import org.elasticsearch.search.aggregations.reducers.derivative.DerivativeParser;
+import org.elasticsearch.search.aggregations.support.AggregationPath;
+import org.elasticsearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class ContiguousBucketHelpers {
```

Comment:
Could probably just call this `BucketHelpers` instead of `ContiguousBucketHelpers`?